src/3d/qgsshadowrenderingframegraph.cpp

```diff
@@ -102,6 +110,38 @@ Qt3DRender::QFrameGraphNode *QgsShadowRenderingFrameGraph::constructForwardRende
 
   mFrustumCulling = new Qt3DRender::QFrustumCulling( forwaredRenderStateSet );
 
+  mForwardClearBuffers = new Qt3DRender::QClearBuffers( mFrustumCulling );
+  mForwardClearBuffers->setClearColor( QColor::fromRgbF( 0.0, 0.0, 1.0, 1.0 ) );
+  mForwardClearBuffers->setBuffers( Qt3DRender::QClearBuffers::ColorDepthBuffer );
+  mForwardClearBuffers->setClearDepthValue( 1.0f );
+
+  Qt3DRender::QLayerFilter *transparentObjectsLayerFilter = new Qt3DRender::QLayerFilter( mForwardRenderTargetSelector );
+  transparentObjectsLayerFilter->addLayer( mTransparentObjectsPassLayer );
+  transparentObjectsLayerFilter->setFilterMode( Qt3DRender::QLayerFilter::AcceptAnyMatchingLayers );
+
+  Qt3DRender::QSortPolicy *sortPolicy = new Qt3DRender::QSortPolicy( transparentObjectsLayerFilter );
+  QVector<Qt3DRender::QSortPolicy::SortType> sortTypes;
+  sortTypes.push_back( Qt3DRender::QSortPolicy::BackToFront );
+  sortTypes.push_back( Qt3DRender::QSortPolicy::Material );
+  sortTypes.push_back( Qt3DRender::QSortPolicy::StateChangeCost );
+  sortTypes.push_back( Qt3DRender::QSortPolicy::FrontToBack );
```

Comment:
It is an order list, if BackToFront ordering returns the same rank in the objects list it will defer to the next sort type and so one. But I believe it doesn't really matter and confusing so I will delete the rest...